test/e2e/generated_clientset.go

```diff
@@ -126,31 +154,83 @@ var _ = KubeDescribe("Generated release_1_2 clientset", func() {
 		}
 
 		By("verifying pod deletion was observed")
-		deleted := false
-		timeout := false
-		var lastPod *api.Pod
-		// The 30s grace period is not an upper-bound of the time it takes to
-		// delete the pod from etcd.
-		timer := time.After(60 * time.Second)
-		for !deleted && !timeout {
-			select {
-			case event, _ := <-w.ResultChan():
-				if event.Type == watch.Deleted {
-					lastPod = event.Object.(*api.Pod)
-					deleted = true
-				}
-			case <-timer:
-				timeout = true
-			}
+		lastPod := observePodDeletion(w)
+		Expect(lastPod.DeletionTimestamp).ToNot(BeNil())
+		Expect(lastPod.Spec.TerminationGracePeriodSeconds).ToNot(BeZero())
+
+		options = api.ListOptions{LabelSelector: selector}
+		pods, err = podClient.List(options)
+		if err != nil {
+			Fail(fmt.Sprintf("Failed to list pods to verify deletion: %v", err))
+		}
+		Expect(len(pods.Items)).To(Equal(0))
+	})
+})
+
+var _ = KubeDescribe("Generated release_1_3 clientset", func() {
+	framework := NewDefaultFramework("clientset")
+	It("should create pods, delete pods, watch pods", func() {
+		podClient := framework.Clientset_1_3.Core().Pods(framework.Namespace.Name)
+		By("creating the pod")
+		name := "pod" + string(util.NewUUID())
+		value := strconv.Itoa(time.Now().Nanosecond())
+		podCopy := testingPod(name, value)
+		pod := &podCopy
+		By("setting up watch")
+		selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": value}))
+		options := api.ListOptions{LabelSelector: selector}
+		pods, err := podClient.List(options)
+		if err != nil {
+			Failf("Failed to query for pods: %v", err)
+		}
+		Expect(len(pods.Items)).To(Equal(0))
+		options = api.ListOptions{
+			LabelSelector:   selector,
+			ResourceVersion: pods.ListMeta.ResourceVersion,
+		}
+		w, err := podClient.Watch(options)
+		if err != nil {
+			Failf("Failed to set up watch: %v", err)
+		}
+
+		By("submitting the pod to kubernetes")
+		// We call defer here in case there is a problem with
+		// the test so we can ensure that we clean up after
+		// ourselves
+		defer podClient.Delete(pod.Name, api.NewDeleteOptions(0))
```

Comment:
I would do the defer after the create, but I guess it doesn't really matter in this case.